src/hotspot/share/runtime/globals.hpp

```diff
@@ -1993,6 +1993,9 @@ const int ObjectAlignmentInBytes = 8;
                                                                             \
   product(bool, StressSecondarySupers, false, DIAGNOSTIC,                   \
           "Use a terrible hash function in order to generate many collisions.") \
+                                                                            \
+  develop(bool, RestrictStable, true,                                       \
```

Comment:
What's the use case for the flag? Solely for testing purposes (since it's develop)? 
Alternatively, you could place the test classes on boot class path and enable test execution with product binaries.

Reply:
Yes, to access the annotation from test, like `RestrictContended` nearby:
https://github.com/openjdk/jdk/blob/97f7c03dd0ff389abefed7ea2a7257bcb42e0754/src/hotspot/share/classfile/classFileParser.cpp#L1960

Not sure if putting test classes on bootclasspath would work well with IR tests that are now running in driver mode.

Reply:
`RestrictContended` and `RestrictReservedStack` are product flags.

Reply:
I'm not saying that `RestrictStable` should be made product. It was a deliberate decision to limit it only to trusted classes. 

There are existing tests for `@Stable` (under `test/hotspot/jtreg/compiler/stable/`) and they don't require any special assistance from the JVM.

Reply:
Again, the problem here is that new tests are *IR Tests*, and I am struggling to find a good way to bootclasspath the classes that IR Test Framework itself invokes. The tests you referenced are not IR tests, and so do not have this problem.

Reply:
IIUC, you want to somehow have `-Xbootclasspath/a:path_to_your_ir_test`. I guess that's currently not so easy to do and probably needs some IR framework support.

I've had a quick look. How about something like that (just prototyped, not fully tested): [ir_framework.patch](https://github.com/user-attachments/files/16471108/ir_framework.log)


Then you could use it like that:
```
        TestFramework testFramework = new TestFramework();
        testFramework
                .addFlags("...", "...")
                .bootstrapTestClasses()
                .start();
```

If that works for all your tests, then you could either add the IR framework changes directly to this patch or we do it in preceding RFE separately.

Reply:
Sounds good, I can take care of it tomorrow. I will ping you in the PR once it is out.

Reply:
Redone the tests to use newly available TestFramework API. Retracted the develop flag. Made sure tests work fine in both fastdebug and release modes. See new commit.